yellowpages/yellowpages-war/src/main/java/org/silverpeas/components/yellowpages/control/YellowpagesSessionController.java

```diff
@@ -765,185 +764,123 @@ public String getCurrentSearchCriteria() {
+      // a contact that is a user that belongs to a group in Silverpeas
+      for (UserFull userFull: listFullUsers) {
+        if (contactFather.getContactDetail().getUserId().equals(userFull.getId())) {
+          result.add(contactFather);
+          break;
         }
-        users = temp;
       }
-
-      // filter users who are in the component
-      TopicDetail rootTopic = getTopic(NodePK.ROOT_NODE_ID);
-      Collection<ContactFatherDetail> contactsUser = getAllContactDetails(rootTopic.getNodePK());
-      Iterator<ContactFatherDetail> iterator = contactsUser.iterator();
-      ContactFatherDetail contactFather;
-      ContactDetail contact;
-      String userId;
-      UserDetail userDetail;
-      while (iterator.hasNext()) {
-        contactFather = iterator.next();
-        contact = contactFather.getContactDetail();
-        userId = contact.getUserId();
-        if (userId != null) {
-          for (UserDetail user : users) {
-            userDetail = user;
-            if (userId.equals(userDetail.getId())) {
-              result.add(contactFather);
-              break;
-            }
-          }
+    } else {
+      // a contact from an internal or external directory
+      for (CompleteContact userComplete: listCompleteUsers) {
+        if (contactFather.getContactDetail().getPK().getId()
+            .equals(userComplete.getContactDetail().getPK().getId())) {
+          result.add(contactFather);
+          break;
         }
       }
+    }
+  }
 
-      // 2 - Look for contacts
-      List<ContactDetail> contacts = new ArrayList<>();
-      if ("LastName".equals(typeSearch)) {
-        contacts = (List<ContactDetail>) getYellowpagesService()
-            .getContactDetailsByLastName(new ContactPK("useless", getComponentId()), nom);
-      } else if ("FirstName".equals(typeSearch)) {
-        contacts = (List<ContactDetail>) getYellowpagesService()
-            .getContactDetailsByLastNameAndFirstName(new ContactPK("useless", getComponentId()),
-                nom, prenom);
-      } else if ("LastNameFirstName".equals(typeSearch)) {
-        if (prenom == null) {
-          // nom ou prenom
-          contacts = (List<ContactDetail>) getYellowpagesService()
-              .getContactDetailsByLastNameOrFirstName(new ContactPK("useless", getComponentId()),
-                  nom);
-        } else {
-          // nom et prenom
-          contacts = (List<ContactDetail>) getYellowpagesService()
-              .getContactDetailsByLastNameAndFirstName(new ContactPK("useless", getComponentId()),
-                  nom, prenom);
-        }
-      }
+  /**
+   * @param typeSearch
+   * @param query
+   * @return list of ContactFatherDetail
+   * @throws FormException
+   * @throws PublicationTemplateException
+   * @
+   */
+  public List<ContactFatherDetail> search(String typeSearch, String query)
+      throws PublicationTemplateException, FormException {
+    List<ContactFatherDetail> result;
+    String _query = query.trim();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00117)